client/utils/rest.go

```diff
@@ -156,40 +156,50 @@ unmarshals to the req interface.
     req := new(SomeReq)
     err := ReadRESTReq(w, r, cdc, req)
 */
-func ReadRESTReq(w http.ResponseWriter, r *http.Request, cdc *codec.Codec, req interface{}) error {
+func ReadRESTReq(w http.ResponseWriter, r *http.Request, cdc *codec.Codec, cliCtx context.CLIContext, req interface{}) (context.CLIContext, error) {
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		WriteErrorResponse(w, http.StatusBadRequest, err.Error())
-		return err
+		return cliCtx, err
 	}
 
 	err = cdc.UnmarshalJSON(body, req)
 	if err != nil {
 		WriteErrorResponse(w, http.StatusBadRequest, err.Error())
-		return err
+		return cliCtx, err
 	}
 
-	return nil
+	genOnlyStr := r.FormValue("generate_only")
+	if len(genOnlyStr) > 0 {
+		cliCtx.GenerateOnly, err = strconv.ParseBool(genOnlyStr)
```

Comment:
previously we only checked if the `generate_only `  had a value, not that it was actually `true` or `false`. 
I think this is also the case for `simulate`. @alessio can you fix that on #3056 ?